### DISCUSSION TITLE
Question: What 'Feature not available' error means?

### DISCUSSION THREAD

**Discussion by lbalusek** [gssapi/gss-ntlmssp#59] [General]
Hello,

I am trying to solve [this issue](https://stackoverflow.com/questions/68084024/asp-net-hosted-on-centos7-with-windows-authentication-gssapi-feature-not-avai), in short, we have ASP.NET server hosted on CentOS 7 and we would like to use Windows authentication, but we are getting the following error

```2021-06-21 07:51:55.4321|ERROR|Microsoft.AspNetCore.Authentication.Negotiate.NegotiateHandler|An exception occurred while processing the authentication request. Interop+NetSecurityNative+GssApiException: GSSAPI operation failed with error - Unspecified GSS failure.  Minor code may provide more information (Feature not available).```

What exactly this error means? 
Thank you very much.

**Comment by simo5** [comment]
Is this related to GSS-NTLMSSP ? (Or the lack thereof?)
In general Feature Not Available means the application tried to perform an operation, but this operation is not supported by the underlying GSSAPI implementation which returned (GSS_S_UNAVAILABLE).

There are many reasons why this may happen, so you need to figure out which GSSAPI call was being made. What GSSAPI mechanism are available on the system (by default only krb5 and the SPNEGO pseudo mechanism), and whether any particular configuration is applied.
